packages/in_app_purchase/lib/src/in_app_purchase_connection/in_app_purchase_connection.dart

```diff
@@ -6,12 +6,19 @@ import 'dart:async';
 import 'dart:io';
 import 'app_store_connection.dart';
 import 'google_play_connection.dart';
+import 'product.dart';
 
 /// Basic generic API for making in app purchases across multiple platforms.
 abstract class InAppPurchaseConnection {
   /// Returns true if the payment platform is ready and available.
   Future<bool> isAvailable();
 
+  /// Query the product list.
+  ///
+  /// Returns a List of [Product] that each [Product] uniquely matches one valid identifier in [identifiers].
+  /// Any item in the `identifiers` that is not specified in App Store Connect or Google Play Console will be ignored.
+  Future<List<Product>> queryProductDetails(List<String> identifiers);
```

Comment:
An alternative could be returning a custom 'Response' class something like `ProductResponse` which contains the list of the product and some errors if any. Any thoughts on this?

Reply:
I like that idea but I don't know what information we would be setting as the response code.

`BillingClient` returns an error code sometimes, and `StoreKit` returns a list of `invalidProductIdentifiers`. How would we combine that in a way that makes sense?

Reply:
That's also what blocking me to introduce the response class.

Reply:
Discussed offline, going to add a response with a list of the not found product IDs.